backend/question-service/src/routes/questionRoutes.ts

```diff
@@ -10,4 +12,8 @@ router.post("/questions", createQuestion);
 
 router.put("/questions/:id", updateQuestion);
 
+router.post("/questionsquery/:currPage", readQuestionsList);
```

Comment:
should it be a `GET` request? since we are reading and not creating questions

Reply:
hmm will get be enough for sending the conditions to search and filter by

Reply:
like if the search string is very long and the person filter by many conditions

Reply:
oh i think the length of the query shouldnt affect anything? the conditions can be specified as a list i think

Reply:
doesnt the url hv a length limit though?

Reply:
Ohh I think I get your concern, I was searching up on this and some do suggest using `POST` instead of `GET` to handle the case of queries with long params (not sure if it's good practice though)

I feel like it's quite unlikely to hit the URL length limit (around 2k characters?) But if we really want to be safe, alternatively we can either
- restrict the length of query params in the frontend (eg. character limit on the search keyword) 
- do error handling in the backend (catch 414 and have the frontend display something similar to page not found)

Reply:
hmm alright i'll change to GET, then it might be better to restrict search character limit + filter categories in the frontend? (i'll still catch 414 just in case but i think just catch 414 only is abit late, like we let the user query with so many conditions already then we tell them that we can't search with that many.) 
is that ok?

Reply:
wait is it possible for you to check if the url too long on the front end instead

Reply:
I think yes but hmm what do we show the user if the url is too long? I think like what you mentioned it's better not to let the user submit a very long query then only inform them that it's too long right

I can do the character restrictions for the search and filter parameters in the frontend if that helps~

Reply:
yea i think if we can guarantee that url will never be too long with restrictions we can forgo the url too long check? otherwise we probably have to put error message like "too many keywords pls try with less" cuz if we just return server error they might try the same thing again.